terms/montgomery_reduction.md

```diff
@@ -0,0 +1,133 @@
+---
+comments: true
+---
+# Montgomery Reduction
+
+This is an efficient method for computing $R = a \cdot b \mod n$ where $a, b, \text{and } n$ are $k$-bit binary number. For our
+purposes, we suppose that $n$ is an odd number.
+
+## Preliminaries
+
+First, we choose $r > n$ such that: $(r, n) = 1$. Because $n$ is odd, we often choose $r = 2^k$.
+
+This technique replaces division by $n$ with division by a power of $2$, , which is easily performed on a computer since numbers are
+represented in binary form.
+
+To achieve this, we need to convert all the numbers we use into Montgomery space. The $n$-residue of $a$ in Montgomery space is given
+by:
+$\bar a = a \cdot r \mod n$
+
+## Montgomery Multiplication
+
+The Montgomery product is defined as:
+$\bar R = \bar a \cdot \bar b \cdot r^{-1} \mod n$, which represents the $n$-residue of $R = a \cdot b \mod n$.
+
+To compute this product, we need to determine $r^{-1}$, which satisfies the property:  
+
+$r \cdot r^{-1} = 1 \mod n$
+$\iff r \cdot r^{-1} - n \cdot n' = 1$
+
+Therefore, we can find $r^{-1}$ and $n'$ using the
+[Extended Euclid algorithm](https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm). We can express $\bar{R}$ as follows:
+
+$\bar R = \bar a \cdot \bar b \cdot r^{-1} \mod n$
```

Comment:
use array to align equations

Reply:
I dont know how to use that? Please give me a link to the example